tests/integration/ha/test_ha.py

```diff
@@ -417,6 +417,113 @@ async def test_freeze_db_process_node_with_elected_cm(
 
 
 @pytest.mark.abort_on_fail
+async def test_restart_db_process_with_elected_cluster_manager(
+    ops_test: OpsTest, c_writes: ContinuousWrites, c_balanced_writes_runner
+) -> None:
+    """Check cluster self-healing & data indexed/read on process restart on CM node."""
+    app = (await app_name(ops_test)) or APP_NAME
+
+    units_ips = get_application_unit_ids_ips(ops_test, app)
+    leader_unit_ip = await get_leader_unit_ip(ops_test, app=app)
+
+    # find unit currently elected cluster manager
+    first_elected_cm_unit_id = await get_elected_cm_unit_id(ops_test, leader_unit_ip)
+
+    # Killing the only instance can be disastrous.
+    if len(ops_test.model.applications[app].units) < 2:
+        await ops_test.model.applications[app].add_unit(count=1)
+        await ops_test.model.wait_for_idle(
+            apps=[app],
+            status="active",
+            timeout=1000,
+            idle_period=IDLE_PERIOD,
+        )
+
+    # restart the opensearch process
+    await send_kill_signal_to_process(ops_test, app, first_elected_cm_unit_id, signal="SIGTERM")
+
+    # verify new writes are continuing by counting the number of writes before and after 5 seconds
+    # should also be plenty for the cluster manager reelection to happen
+    writes = await c_writes.count()
+    time.sleep(5)
+    more_writes = await c_writes.count()
+    assert more_writes > writes, "writes not continuing to DB"
+
+    # verify that the opensearch service is back running on the unit previously elected CM unit
+    assert await is_up(
+        ops_test, units_ips[first_elected_cm_unit_id]
+    ), "OpenSearch service hasn't restarted."
+
```

Comment:
Please add a check that a new cluster manager has been elected, see [here](https://github.com/canonical/opensearch-operator/blob/main/tests/integration/ha/test_ha.py#L249-L253)